ivy/functional/frontends/paddle/nn/functional/activation.py

```diff
@@ -1,9 +1,34 @@
 # local
+import ivy
+from ivy.func_wrapper import with_supported_dtypes
+from ivy.functional.frontends.paddle.func_wrapper import to_ivy_arrays_and_back
 from ivy.functional.frontends.paddle.tensor.math import tanh as paddle_tanh
 from ivy.functional.frontends.paddle.tensor.math import (
     log_softmax as paddle_log_softmax,
 )
 
 
+@with_supported_dtypes({"2.4.2 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def selu(
+    x,
+    /,
+    *,
+    alpha=1.6732632423543772848170429916717,
+    scale=1.0507009873554804934193349852946,
+    name=None,
+):
+    if scale <= 1.0:
+        raise ValueError(f"The scale must be greater than 1.0. Received: {scale}.")
+
+    if alpha < 0:
+        raise ValueError(f"The alpha must be no less than zero. Received: {alpha}.")
+
+    ret = ivy.where(x > 0, x, alpha * ivy.expm1(x))
+    dtype = ivy.dtype(x)
```

Comment:
No need to call this here

Reply:
How will I infer the dtype of `x` so that I can cast it to `arr` ?

Reply:
Oh yeah you can simply pass in `x.dtype` to `ivy.astype(...)`. On a second thought, I guess this is fine as well. Let's leave it be :)